examples/02_decoding/plot_haxby_understand_decoder.py

```diff
@@ -0,0 +1,210 @@
+# Under the hood, it uses :class:`nilearn.maskers.NiftiMasker` to do all these
+# operations. So here we will demonstrate this by directly using the
+# :class:`nilearn.maskers.NiftiMasker`. We will use it to:
+#   1. only keep the data from the Ventral Temporal cortex by providing the
+#      mask image (in :class:`nilearn.decoding.Decoder` this is done by
+#      providing the mask image in the ``mask`` parameter).
+#   2. standardize the data by z-scoring it such that the data is scaled to
+#      have zero mean and unit variance across trials (in
+#      :class:`nilearn.decoding.Decoder` this is done by setting the
+#      ``standardize`` parameter to ``"zscore_sample"``).
+
+from nilearn.maskers import NiftiMasker
+
+masker = NiftiMasker(mask_img=mask_vt, standardize="zscore_sample")
+
+# Fit and transform the data with the masker
+X = masker.fit_transform(fmri_img)
+
+print(f"fMRI data shape after masking: {X.shape}")
+# So now we have a 2D numpy array of shape (864, 464) where each row
+# corresponds to a trial and each column corresponds to a feature
+# (voxel in the Ventral Temporal cortex).
+
+# %%
+# Convert the multi-class labels to binary labels
+# -----------------------------------------------
+#
+# The :class:`nilearn.decoding.Decoder` converts multi-class classification
+# problem to N one-vs-others binary classification problems by default, where N
+# is the number of unique labels. We have 8 unique labels in our case.
+
+from matplotlib import pyplot as plt
+from sklearn.preprocessing import LabelBinarizer
+
+label_binarizer = LabelBinarizer(pos_label=1, neg_label=-1)
+y_binary = label_binarizer.fit_transform(y)
+
+# %%
+# Let's plot the labels to understand the conversion
+from matplotlib.colors import ListedColormap
+from sklearn.preprocessing import LabelEncoder
+
+# create a copy of y_binary and manipulate it just for plotting
+y_binary_ = y_binary.copy()
+for col in range(y_binary_.shape[1]):
+    y_binary_[np.where(y_binary_[:, col] == 1), col] = col
+
+fig, (ax_binary, ax_multi) = plt.subplots(
+    2, gridspec_kw={"height_ratios": [10, 1.5]}, figsize=(12, 2)
+)
+cmap = ListedColormap(["white"] + list(plt.cm.tab10.colors)[0:8])
+binary_plt = ax_binary.imshow(
+    y_binary_.T,
+    aspect="auto",
+    cmap=cmap,
+    interpolation="nearest",
+    origin="lower",
+)
+ax_binary.set_xticks([])
+ax_binary.set_yticks([])
+ax_binary.set_ylabel("One-vs-Others")
+# encode the original labels for plotting
+label_multi = LabelEncoder()
+y_multi = label_multi.fit_transform(y)
+y_multi = y_multi.reshape(1, -1)
+cmap = ListedColormap(list(plt.cm.tab10.colors)[0:8])
+multi_plt = ax_multi.imshow(
+    y_multi,
+    aspect="auto",
+    interpolation="nearest",
+    cmap=cmap,
+)
+ax_multi.set_yticks([])
+ax_multi.set_xlabel("Original trial sequence")
+cbar = fig.colorbar(multi_plt, ax=[ax_binary, ax_multi])
+cbar.set_ticks(np.arange(1 + len(label_multi.classes_)))
+cbar.set_ticklabels([*label_multi.classes_, "all others"])
+
+plt.show()
```

Comment:
To explain how the One-vs-Other label encoding works
![binary_classes](https://github.com/nilearn/nilearn/assets/35485429/1b29d0c6-0149-47e2-ab31-9045ee0732de)